### PR TITLE
Homework10 git

### PR DESCRIPTION
**Pull request by Iliailineya**
Implement with controllers: User, account, payments
Implement with services: User, account, payments
Implement with repositories: User, account, payments
Add Exception handlers with controller advice
Additional task:

Implement integration test for controllers

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -3,6 +3,7 @@ target/
 !.mvn/wrapper/maven-wrapper.jar
 !**/src/main/**/target/
 !**/src/test/**/target/
+src/main/resources/app.properties
```
Comment (YeaaaaH) [inline .gitignore]: same here

File: src/main/java/spring/repository/AccountRepository.java [hunk 2]
```diff
@@ -0,0 +1,9 @@
+package spring.repository;
+
+import spring.model.Account;
+import org.springframework.data.jpa.repository.JpaRepository;
+import org.springframework.stereotype.Repository;
+
+@Repository
```
Comment (YeaaaaH) [inline src/main/java/spring/repository/AccountRepository.java]: remove it

File: src/main/java/spring/service/UserService.java [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+        return repository.findById(id)
+                .orElseThrow(() -> new UserNotFoundException("User with id " + id + " not found"));
+    }
+
+    public List<User> getAllUsers() {
+        return repository.findAll();
+    }
+
+    public User updateUser(long id, User user) {
+        if (repository.existsById(id)) {
```
Comment (YeaaaaH) [inline src/main/java/spring/service/UserService.java]: you can use  getUserById()

File: src/main/java/spring/service/PaymentService.java [hunk 4]
```diff
@@ -0,0 +1,48 @@
...
+        return repository.findById(id)
+                .orElseThrow(() -> new PaymentNotFoundException("Payment with id " + id + " not found"));
+    }
+
+    public List<Payment> getAllPayments() {
+        return repository.findAll();
+    }
+
+    public Payment updatePayment(long id, Payment payment) {
+        if (repository.existsById(id)) {
```
Comment (YeaaaaH) [inline src/main/java/spring/service/PaymentService.java]: you can use getPaymentById

**Review by YeaaaaH — CHANGES_REQUESTED** [review]
small fixs needed

File: .gitignore (hunk 1 above)
Comment (Iliailineya) [reply .gitignore]: done

File: src/main/java/spring/repository/AccountRepository.java (hunk 2 above)
Comment (Iliailineya) [reply src/main/java/spring/repository/AccountRepository.java]: done

File: src/main/java/spring/service/PaymentService.java (hunk 4 above)
Comment (Iliailineya) [reply src/main/java/spring/service/PaymentService.java]: done

File: src/main/java/spring/service/UserService.java (hunk 3 above)
Comment (Iliailineya) [reply src/main/java/spring/service/UserService.java]: done

**Comment by Iliailineya** [comment]
Fixed

File: src/main/java/spring/service/PaymentService.java [hunk 5]
```diff
@@ -28,13 +29,9 @@ public List<Payment> getAllPayments() {
...
     public Payment updatePayment(long id, Payment payment) {
-        if (repository.existsById(id)) {
-            payment.setId(id);
-            repository.save(payment);
-            return payment;
-        } else {
-            throw new PaymentNotFoundException("Payment with id " + id + " not found");
-        }
+        getPaymentById(id);
+        payment.setId(id);
```
Comment (YeaaaaH) [inline src/main/java/spring/service/PaymentService.java]: this line is redundant, if entity is present in db it already has id

File: src/main/java/spring/service/UserService.java [hunk 6]
```diff
@@ -28,13 +28,9 @@ public List<User> getAllUsers() {
...
     public User updateUser(long id, User user) {
-        if (repository.existsById(id)) {
-            user.setId(id);
-            repository.save(user);
-            return user;
-        } else {
-            throw new UserNotFoundException("User with id " + id + " not found");
-        }
+        getUserById(id);
+        user.setId(id);
```
Comment (YeaaaaH) [inline src/main/java/spring/service/UserService.java]: this line is redundant, if entity is present in db it already has id

File: src/main/java/spring/service/AccountService.java [hunk 7]
```diff
@@ -28,13 +28,9 @@ public List<Account> getAllAccounts() {
...
     public Account updateAccount(long id, Account account) {
-        if (repository.existsById(id)) {
-            account.setId(id);
-            repository.save(account);
-            return account;
-        } else {
-            throw new AccountNotFoundException("Account with id " + id + " not found");
-        }
+        getAccountById(id);
+        account.setId(id);
```
Comment (YeaaaaH) [inline src/main/java/spring/service/AccountService.java]: this line is redundant, if entity is present in db it already has id

**Review by YeaaaaH — CHANGES_REQUESTED** [review]
need to fix

**Comment by Iliailineya** [comment]
Fixed

**Pull request merged**